lib/quantified/coder.rb

```diff
@@ -0,0 +1,22 @@
+module Quantified
+  class Coder
+    attr_reader :attribute, :unit
+
+    def initialize(attribute, unit)
+      @attribute = attribute
+      @unit = unit
+    end
+
+    def load(amount)
+      attribute.new(amount, unit) unless amount.nil?
+    end
+
+    def dump(quantity)
+      if quantity.is_a?(attribute)
```

Comment:
I wanted to check that the unit type matched. E.g.: not setting a Mass to a Length. But I think you're right that `responds_to?` is the better choice here.